download.sh

```diff
@@ -12,35 +12,39 @@ rm -v artifacts.zip
 mv archive/* ./
 rmdir archive
 
-# remove "sbuild" tarballs
-# we don't use these in Docker, and as of 2017-09-07 unstable/testing are larger than GitHub's maximum file size of 100MB (~140MB)
-# they're still available in the Jenkins artifacts directly for folks who want them (and buildable reproducibly via debuerreotype)
-rm -rf */sbuild/
-
-# remove empty files (temporary fix for https://github.com/debuerreotype/debuerreotype/commit/d29dd5e030525d9a5d9bd925030d1c11a163380c)
-find */ -type f -empty -delete
-
 snapshotUrl="$(cat snapshot-url 2>/dev/null || echo 'https://deb.debian.org/debian')"
 dpkgArch="$(< dpkg-arch)"
 
 for suite in */; do
 	suite="${suite%/}"
 
-	[ -f "$suite/rootfs.tar.xz" ]
-	cat > "$suite/Dockerfile" <<-'EODF'
-		FROM scratch
-		ADD rootfs.tar.xz /
-		CMD ["bash"]
-	EODF
-	# TODO cleverly detect whether "bash" exists in "rootfs.tar.xz" (and fall back to "sh" if not)
-	# https://salsa.debian.org/debian/grow-your-ideas/-/issues/20
-	cat > "$suite/.dockerignore" <<-'EODI'
-		**
-		!rootfs.tar.xz
-	EODI
+	for variant in '' slim; do
+		dir="$suite${variant:+/$variant}"
+		[ -d "$dir" ]
+		[ -s "$dir/oci.tar" ]
+
+		mkdir "$dir/oci"
+		tar -xf "$dir/oci.tar" -C "$dir/oci"
+		rm -f "$dir/oci.tar" "$dir/rootfs.tar.xz"
+		rootfs='oci/blobs/rootfs.tar.gz'
+
+		[ -s "$dir/$rootfs" ]
+
+		cmd="$(jq -c '.config.Cmd' "$dir/oci/blobs/image-config.json")"
+		[[ "$cmd" = '["'*'"]' ]]
+
+		# this won't be used by us anymore, but included for backwards compatibility
```

Comment:
Could you move this comment inside the Dockerfile for visibility?

Reply:
Great idea, updated!